File: back_streaming/src/main/java/com/sovereingschool/back_streaming/Models/Plan.java
package com.sovereingschool.back_streaming.Models;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "plan")
public class Plan implements Serializable {
    @Id
    private Long id_plan;

    private String nombre_plan;

    private BigDecimal precio_plan;

    private List<Curso> cursos_plan;

}

File: front/src/app/models/Plan.ts
export class Plan {
	public id_plan: number;

	public nombre_plan?: string | null;

	public precio_plan?: number | null;

	constructor(_id_plan: number, _nombre_plan?: string, _precio_plan?: number) {
		this.id_plan = _id_plan;
		this.nombre_plan = _nombre_plan;
		this.precio_plan = _precio_plan;
	}
}
